lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +103,188 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
+    // the end of this method.
 
-    Instant endTime = Instant.now();
+    // Run a pause operation.
+    PausedDuration pausedDuration = null;
+    PauseFailedException pauseFailedException = null;
+    try {
+      pausedDuration = pauseInternal(requestCoordinator, pauseDuration, maxPauseWaitTime);
+    } catch (Exception e) {
+      pauseFailedException = new PauseFailedException(PAUSE_ERROR_MESSAGE, e);
+    }
 
-    unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+    // Run an unpause operation.
+    UnpauseFailedException unpauseFailedException = null;
+    try {
+      unpauseWithRetry(requestCoordinator, MAX_UNPAUSE_RETRY_COUNT);
+    } catch (Exception e) {
+      unpauseFailedException = new UnpauseFailedException(UNPAUSE_ERROR_MESSAGE, e);
+    }
 
-    TargetSnapshot targetAfterPause;
+    // Get pods and deployment information after pause.
+    TargetSnapshot targetAfterPause = null;
+    GetTargetAfterPauseFailedException getTargetAfterPauseFailedException = null;
     try {
-      targetAfterPause = targetSelector.select();
+      targetAfterPause = getTarget();
     } catch (Exception e) {
-      throw new PauserException(
-          "Failed to find the target pods to examine if the targets pods were updated during"
-              + " paused.",
-          e);
+      getTargetAfterPauseFailedException =
+          new GetTargetAfterPauseFailedException(GET_TARGET_AFTER_PAUSE_ERROR_MESSAGE, e);
     }
 
-    if (!target.getStatus().equals(targetAfterPause.getStatus())) {
-      throw new PauserException("The target pods were updated during paused. Please retry.");
+    // Check if pods and deployment information are the same between before pause and after pause.
+    StatusCheckFailedException statusCheckFailedException = null;
+    StatusUnmatchedException statusUnmatchedException = null;
+    try {
+      statusUnmatchedException = targetStatusEquals(targetBeforePause, targetAfterPause);
+    } catch (Exception e) {
+      statusCheckFailedException = new StatusCheckFailedException(STATUS_CHECK_ERROR_MESSAGE, e);
     }
 
-    return new PausedDuration(startTime, endTime);
+    // We use the exceptions as conditions instead of using boolean flags like `isPauseOk`, etc. If
+    // we use boolean flags, it might cause a bit large number of combinations. For example, if we
+    // have three flags, they generate 2^3 = 8 combinations. It also might make the nested if
+    // statement or a lot of branches of the switch statement. That's why we don't use status flags
+    // for now.
+    PauserException pauserException =
+        buildException(
+            unpauseFailedException,
+            pauseFailedException,
+            getTargetAfterPauseFailedException,
+            statusCheckFailedException,
+            statusUnmatchedException);
+
+    // Return the final result based on each process.
+    if (pauserException != null) {
+      // Some operations failed.
+      throw pauserException;
```

Comment:
[minor] Should we have some log messages when `pauseException != null`? 👀 

Looks like we removed a log in this PR (Sorry if I miss some context here 😓)
```java
          logger.warn(
              "Failed to unpause Scalar product. They are still in paused. You must restart related"
                  + " pods by using the `kubectl rollout restart deployment {}`"
                  + " command to unpase all pods.",
              target.getDeployment().getMetadata().getName());
          return;
```

Reply:
I think we don't need to output such a log message here.

I decided to include all messages in the generated exception.

Now, we can show users clearer errors by using several kinds of exceptions (e.g., `UnpauseFailedException`).

Also, I wanted to see `What happened` from the log, but now we can see all the messages in the exceptions via the stacktrace.

So, there is no problem if we don't have a log here.